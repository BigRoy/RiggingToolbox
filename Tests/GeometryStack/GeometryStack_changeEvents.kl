
require RiggingToolbox;

operator entry(){
   	
   	GeometryStack stack();
   	PolygonMeshSphereGenerator sphereGenerator(2.0, 1, true, true);
   	PushDeformer pushDeformer(3.0);
   	sphereGenerator.enableCaching = true;
   	stack.addGeometryOperator(sphereGenerator);
   	stack.addGeometryOperator(pushDeformer);
   	stack.initCachePoints();

    EvalContext context();
    {
	    Geometry geometries[] = stack.evaluate(context);

	    report("NumGeometries:" + geometries.size);
	    for(Integer i=0; i<geometries.size; i++){
	        report(geometries[i].getDesc(true, true));
	    }
	}
	pushDeformer.setPushDist(6.0);
	{
	    Geometry geometries[] = stack.evaluate(context);

	    report("NumGeometries:" + geometries.size);
	    for(Integer i=0; i<geometries.size; i++){
	        report(geometries[i].getDesc(true, true));
	    }
	}
}

