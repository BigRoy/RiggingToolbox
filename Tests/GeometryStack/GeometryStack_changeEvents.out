function GeometrySet Ref<GeometryStack>.evaluate!(EvalContext)
--function Ref<PolygonMeshSphereGenerator>#evaluate?(EvalContext, io GeometrySet)
--function Ref<GeometryCache>#init!(GeometrySet)
--function Ref<PushDeformer>#evaluate?(EvalContext, io GeometrySet)
--function Ref<GeometryAttributeCache>#init!(GeometrySet)
function Ref<GeometryStack>#notify!(Notifier, String, String):changed
function GeometrySet Ref<GeometryStack>.evaluate!(EvalContext)
--function Ref<GeometryCache>#restore?(io GeometrySet)
--function Ref<PushDeformer>#evaluate?(EvalContext, io GeometrySet)
--function Ref<GeometryAttributeCache>#init!(GeometrySet)

